pkg/proxy/proxy/transparent.go

```diff
@@ -175,6 +203,55 @@ func (t *TransparentProxyService) ServeMetadata(addr string) {
 	}
 }
 
+// Check that the requested url is allowed by the network policy.
+func (proxy TransparentProxyService) CheckNetworkPolicy(req *http.Request, ctx *goproxy.ProxyCtx) (*http.Request, *http.Response) {
+	if proxy.Mode != EnforcementMode {
+		return req, nil
+	}
+	url := req.URL
+	if proxy.Policy != nil {
+		for _, rule := range proxy.Policy.Rules {
+			if url.Hostname() != rule.Host {
+				continue
+			}
+			if urlSatisfiesRule(url, rule) {
+				return req, nil
+			}
+		}
+	}
+	log.Printf("Request to %s blocked by network policy", url.String())
+	errorMessage := fmt.Sprintf("Access to %s is blocked by the proxy's network policy", url.String())
+	return req, goproxy.NewResponse(req, goproxy.ContentTypeText, http.StatusForbidden, errorMessage)
+}
+
+type MatchingStrategy int
+
+const (
+	PathPrefix MatchingStrategy = iota
```

Comment:
my main concern would be that prefix is a more permissive default so the failure/misuse case is worse than with full match.

Reply:
Thinking about it again I think if the user gets to this point they are probably intentionally trying to enforce a policy, in which case I agree it should probably be more restrictive than not. 

I've changed this to a string enum instead (`MatchingType` in policy.go) so that the type must explicitly match one of these options. wdyt?